MoreLinq/Lag.cs

```diff
@@ -36,9 +37,13 @@ public static partial class MoreEnumerable
         /// <param name="resultSelector">A projection function which accepts the current and lagged items (in that order) and returns a result</param>
         /// <returns>A sequence produced by projecting each element of the sequence with its lagged pairing</returns>
 
-        public static IEnumerable<TResult> Lag<TSource, TResult>(this IEnumerable<TSource> source, int offset, Func<TSource, TSource, TResult> resultSelector)
+        public static IEnumerable<TResult> Lag<TSource, TResult>(this IEnumerable<TSource> source, int offset, Func<TSource, TSource?, TResult> resultSelector)
         {
-            return Lag(source, offset, default!, resultSelector);
+            if (source == null) throw new ArgumentNullException(nameof(source));
```

Comment:
Consistency: `is null` vs `== null`. Probably prefer `is null`, but I'm good either way.